pkg/apis/eksctl.io/v1alpha5/defaults.go

```diff
@@ -62,14 +62,16 @@ func SetNodeGroupDefaults(ng *NodeGroup, meta *ClusterMeta) {
 		}
 	}
 
+	if ng.SSH.Allow == nil {
+		ng.SSH.Allow = Disabled()
+	}
+
 	numSSHFlagsEnabled := countEnabledFields(
 		ng.SSH.PublicKeyName,
 		ng.SSH.PublicKeyPath,
 		ng.SSH.PublicKey)
 
-	if numSSHFlagsEnabled > 0 {
```

Comment:
```suggestion
	// Enable when an ssh key is passed and ssh access is not explicitly disabled
	if numSSHFlagsEnabled > 0 && !IsDisabled(ng.SSH.Allow) {
```

Reply:
Hello @martina-if yeah I think that change makes sense and would work. I'll look at implementing that tomorrow at work. Thanks for the response :)